src/vm/runners/builtin_runner/bitwise.rs

```diff
@@ -98,6 +98,24 @@ impl BitwiseBuiltinRunner {
     pub fn get_memory_segment_addresses(&self) -> (&'static str, (isize, Option<usize>)) {
         ("bitwise", (self.base, self.stop_ptr))
     }
+
+    pub fn get_used_diluted_check_units(&self, diluted_spacing: u32, diluted_n_bits: u32) -> usize {
+        let total_n_bits = self.total_n_bits;
+        let mut partition = Vec::with_capacity(total_n_bits as usize);
+        for i in (0..total_n_bits).step_by((diluted_spacing * diluted_n_bits) as usize) {
+            for j in 0..diluted_spacing {
+                if i + j < total_n_bits {
+                    partition.push(i + j)
+                };
+            }
+        }
+        let partition_lengh = partition.len();
+        let num_trimmed = partition
+            .into_iter()
+            .filter(|elem| elem + diluted_spacing * (diluted_n_bits - 1) + 1 > total_n_bits)
+            .count();
+        4 * partition_lengh + num_trimmed
+    }
```

Comment:
I think this merits a comment explaining what it does. It's certainly not obvious.

Reply:
The only documentation for this method is "Returns the number of diluted check units used by the builtin." in the class' method definition, so Im not really sure what it does either

Reply:
There's a tiny bit more context in the callers of this function, apparently those "diluted check units" are extra zeros in between information carrying bits. I'm guessing some error correcting code gets injected in those at some later point.